app/main.py

```diff
@@ -1,6 +1,15 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    archive_functions = {}
+
+    def wrapper(*args, **kwargs) -> Any:
```

Comment:
You could use a decorator from `functools` for the `wrapper` to preserve the decorated function name and docstring.